src/Schema/Builder.php

```diff
@@ -146,6 +151,70 @@ public function getTableListing()
         return $collections;
     }
 
+    public function getColumns($table)
```

Comment:
This method could get slow for larger collections, so it may be worth adding a [`$sample`](https://www.mongodb.com/docs/manual/reference/operator/aggregation/sample/) pipeline stage down the road.

Depending on how much the feature is used, we can also consider expanding its functionality in a separate project to be more similar to [Compass' schema analysis feature](https://www.mongodb.com/docs/compass/current/schema/). For now, the current solution provides a good starting point.

Reply:
I don't think we should try to replicate what is done by Compass. I added `$sample` and `$limit` stages.

Reply:
I tried on atlas sample dataset ``. With `$sample`, I get this error:
```
  PlanExecutor error during aggregation :: caused by :: Sort exceeded memory limit of 33554432 bytes, but did not opt in to external sorting. Aborting operation. Pass allowDiskUse:true to opt in.
```

It works without it. There might be an optimization on the server for this use-case.